### ISSUE TITLE
[1.9.2] Problem with .bans

### ISSUE THREAD

**Issue by wilkowy** [eggheads/eggdrop#1297]
Eggdrop 1.9.2

There is problem with command `.bans` and probably `.exempts` and `.invites` as well.

The test were carried out on a busy bot with many users, bans and a few channels. Let's assume we have such environment: we have some global bans; `#first_chan` is a channel on top op `eggdrop.chan` so this is the first channel in `.status`; `#chan_with_some_bans` which has a test timed-ban for the test purposes; and `#chan_with_no_ban` is the channel with no local bans.

Now the test (mind using the global console channel):

- `.console #chan_with_some_bans`
```
Set your console to #chan_with_some_bans: mjkcoblxsw (msgs, joins, kicks/modes, cmds, misc, bots, linked bot msgs, files, server input, wallops).
```

`.bans`
```
Global bans:
Channel bans for #chan_with_some_bans:  (* = not placed by bot)
Use '.bans all' to see the total list.
```

`.bans all`
```
Global bans:   (! = not active on #chan_with_some_bans)
[...]
Channel bans for #chan_with_some_bans:   (! = not active, * = not placed by bot)
[...]
```

`.bans #chan_with_some_bans`
```
Global bans:
Channel bans for #chan_with_some_bans:  (* = not placed by bot)
Use '.bans all' to see the total list.
```

`.bans #first_chan`
```
Global bans:
Channel bans for #first_chan:  (* = not placed by bot)
[...]
Use '.bans all' to see the total list.
```

- `.console #chan_with_no_ban`
```
Set your console to #chan_with_no_ban: mjkcoblxsw (msgs, joins, kicks/modes, cmds, misc, bots, linked bot msgs, files, server input, wallops).
```

`.bans`
```
Global bans:
Channel bans for #chan_with_no_ban:  (* = not placed by bot)
Use '.bans all' to see the total list.
```

`.bans all`
```
Global bans:   (! = not active on #chan_with_no_ban)
[...]
Channel bans for #chan_with_no_ban:   (! = not active, * = not placed by bot)
```

`.bans #chan_with_some_bans`
```
Global bans:
Channel bans for #chan_with_some_bans:  (* = not placed by bot)
Use '.bans all' to see the total list.
```

`.bans #first_chan`
```
Global bans:
Channel bans for #first_chan:  (* = not placed by bot)
[...]
Use '.bans all' to see the total list.
```

- `.console #first_chan`
```
Set your console to #first_chan: mjkcoblxsw (msgs, joins, kicks/modes, cmds, misc, bots, linked bot msgs, files, server input, wallops).
```

`.bans`
```
Global bans:
Channel bans for #first_chan:  (* = not placed by bot)
[...]
Use '.bans all' to see the total list.
```

`.bans all`
```
Global bans:   (! = not active on #first_chan)
[...]
Channel bans for #first_chan:   (! = not active, * = not placed by bot)
[...]
```

`.bans #chan_with_some_bans`
```
Global bans:
Channel bans for #chan_with_some_bans:  (* = not placed by bot)
Use '.bans all' to see the total list.
```

`.bans #first_chan`
```
Global bans:
Channel bans for #first_chan:  (* = not placed by bot)
[...]
Use '.bans all' to see the total list.
```

- `.console *`
```
Your console is *: mjkcoblxsw (msgs, joins, kicks/modes, cmds, misc, bots, linked bot msgs, files, server input, wallops).
```

`.bans`
```
Global bans:
Channel bans for #first_chan:  (* = not placed by bot)
[...]
Use '.bans all' to see the total list.
```

`.bans all`
```
Global bans:   (! = not active on #first_chan)
[...]
Channel bans for #first_chan:   (! = not active, * = not placed by bot)
[...]
```

`.bans #chan_with_some_bans`
```
Global bans:
Channel bans for #chan_with_some_bans:  (* = not placed by bot)
Use '.bans all' to see the total list.
```

`.bans #first_chan`
```
Global bans:
Channel bans for #first_chan:  (* = not placed by bot)
[...]
Use '.bans all' to see the total list.
```

As you can see this code does not work..

**Comment by vanosg** [comment]
Hi @wilkowy  ... sorry, its still early here- could you briefly summarize what you expect to happen and what is not happening? Thanks a ton

**Comment by vanosg** [comment]
Hi @wilkowy - we want to look into this, but still not sure what the specific problem you're identifying is. Can you rephrase it, so we know where to focus our attention? Otherwise we'll have to close this out. Thanks for the help!

**Comment by wilkowy** [comment]
The issue is that when `.console *` is used then `.bans`/exempts/invites commands take the first channel from the userlist. The expected behavior would be to iterate through all channels and list their active bans or ignore the command.

Also when using `.bans #channel` (on different console channel) it is not possible to show all bans despite _"Use '.bans all' to see the total list."_ line.

**Comment by vanosg** [comment]
thanks for the additional clarification!
